src/babashka/fs.cljc

```diff
@@ -1125,3 +1128,51 @@
   (if win?
     (-> f as-path .toUri .getPath)
     (str f)))
+
+(defn- xdg-path-from-env-var
+  "Yields value of environment variable `k` as path if it's an absolute
+  path. Else `nil`."
+  [k]
+  (some-> (get-env k)
+          (#(when (absolute? %) %))
+          (path)))
+
+(def ^:private xdg-type->env-var&default-path
+  {:config ["XDG_CONFIG_HOME" (path (home) ".config")]
```

Comment:
Can you save this result in a delay and only realize it when one of the xdg functions is called? 

Capturing these paths on the top level will not work with graalvm as it will capture the CI environment's paths.

Reply:
Amended!